alpha-commons/src/main/java/at/ac/tuwien/kr/alpha/commons/literals/BasicLiteralImpl.java

```diff
@@ -25,51 +25,55 @@
  * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
-package at.ac.tuwien.kr.alpha.common.atoms;
-
-import at.ac.tuwien.kr.alpha.common.terms.Term;
-import at.ac.tuwien.kr.alpha.common.terms.VariableTerm;
-import at.ac.tuwien.kr.alpha.grounder.Substitution;
+package at.ac.tuwien.kr.alpha.commons.literals;
 
 import java.util.Collections;
 import java.util.HashSet;
 import java.util.Set;
 
+import at.ac.tuwien.kr.alpha.api.grounder.Substitution;
+import at.ac.tuwien.kr.alpha.api.programs.atoms.BasicAtom;
+import at.ac.tuwien.kr.alpha.api.programs.literals.BasicLiteral;
+import at.ac.tuwien.kr.alpha.api.programs.literals.Literal;
+import at.ac.tuwien.kr.alpha.api.terms.Term;
+import at.ac.tuwien.kr.alpha.api.terms.VariableTerm;
+import at.ac.tuwien.kr.alpha.commons.atoms.AbstractAtom;
+
 /**
- * Contains a potentially negated {@link BasicAtom}.
- * 
+ * Contains a potentially negated {@link BasicAtomImpl}.
+ *
  * Copyright (c) 2017-2018, the Alpha Team.
  */
-public class BasicLiteral extends Literal {
-	
-	public BasicLiteral(BasicAtom atom, boolean positive) {
+class BasicLiteralImpl extends AbstractLiteral implements BasicLiteral { // TODO could we parameterize Literal with Atom type?
```

Comment:
The TODO is a discussion item for an issue, not the actual code here.
```suggestion
class BasicLiteralImpl extends AbstractLiteral implements BasicLiteral {
```

Reply:
Mentioned in #300